_source/_posts/2021-12-06-spring-data-elasticsearch.md

```diff
@@ -0,0 +1,338 @@
+
+```yml
+blog:
+  image: blog
+  environment:
+    ...
+    - SPRING_DATA_ELASTICSEARCH_CLIENT_REACTIVE_ENDPOINTS=blog-elasticsearch:9200
+    ...
+```
+
+_The operations abstraction_
+
+`ReactiveElasticsearchOperations` is the gateway for executing high-level commands against an Elasticsearch cluster using the `ReactiveElasticsearchClient`.
+The `ReactiveElasticsearchTemplate` is the default implementation of `ReactiveElasticsearchOperations`. Some high-level operations are `save`, `get`, `delete` and `search`, which accept the Spring Data Query types as parameters.
+
+_The entity search repository_
+
+The entity search repository enables the document CRUD and search operations in the application. The search repository extends `ReactiveElasticsearchRepository`, the persistence technology-specific abstraction, that builds on the core repository support utilizing operations provided through the `ReactiveElasticsearchTemplate` executed by the `ReactiveElasticsearchClient`. JHipster generates a search repository for the entities specified in the `search` option in the JDL. In the example, all entities are set to be searchable in Elasticsearch:
+
+```
+application {
+  config {
+    baseName blog
+    reactive true
+    ...
+    searchEngine elasticsearch
+  }
+  ...
+  search * with elasticsearch
+}
+```
+
+When the entity in the JDL definition requires the Elasticsearch engine, and also pagination, the entity search repository is generated with pagination and sorting options. For example, the generated TagSearchRepository adds a custom `search()` method that assembles a native search query and passes it to the `ReactiveElasticsearchTemplate`:
+
+```java
+public interface TagSearchRepository extends ReactiveElasticsearchRepository<Tag, String>, TagSearchRepositoryInternal {}
+
+interface TagSearchRepositoryInternal {
+    Flux<Tag> search(String query, Pageable pageable);
+}
+
+class TagSearchRepositoryInternalImpl implements TagSearchRepositoryInternal {
+
+    private final ReactiveElasticsearchTemplate reactiveElasticsearchTemplate;
+
+    TagSearchRepositoryInternalImpl(ReactiveElasticsearchTemplate reactiveElasticsearchTemplate) {
+        this.reactiveElasticsearchTemplate = reactiveElasticsearchTemplate;
+    }
+
+    @Override
+    public Flux<Tag> search(String query, Pageable pageable) {
+        List<FieldSortBuilder> builders = new SortToFieldSortBuilderConverter().convert(pageable.getSort());
+
+        NativeSearchQueryBuilder queryBuilder = new NativeSearchQueryBuilder()
+            .withQuery(queryStringQuery(query))
+            .withPageable(PageRequest.of(pageable.getPageNumber(), pageable.getPageSize()));
+
+        builders
+            .stream()
+            .forEach(builder -> {
+                queryBuilder.withSort(builder);
```

Comment:
This looks to be deprecated in the latest version. It seems that `withSorts()` is its replacement. Any chance you can create a PR for JHipster to fix it?

Reply:
Will do after sending this post revisions.